nemo_rl/algorithms/distillation.py

```diff
@@ -731,17 +731,30 @@ def distillation_train(
                         del distillation_save_state["val_reward"]
                     distillation_save_state["consumed_samples"] = consumed_samples
 
-                    if master_config["checkpointing"]["metric_name"] is not None:
-                        if (
-                            master_config["checkpointing"]["metric_name"]
-                            not in distillation_save_state
-                        ):
+                    full_metric_name = master_config["checkpointing"]["metric_name"]
+                    if full_metric_name is not None:
+                        assert full_metric_name.count(":") == 1, (
+                            "metric_name must contain exactly one colon"
+                        )
```

Comment:
maybe a more descriptive error since there are more constraints than just `:`. disclaimer, i didn't test or lint this, but maybe something like this
```suggestion
                        assert full_metric_name.startswith('train:') or full_metric_name.startswith('val:'), (
                            f"metric_name={full_metric_name} must start with 'val:' or 'train:'"
                        )
```

same with the other algos